### ISSUE TITLE
[Bug]: Release(Linux) workflow fails if cache is disabled

### ISSUE THREAD

**Issue by kaomoneus** [OpenTTD/OpenTTD#11356]
### Version of OpenTTD

release 13.4, master (94167dfd34716dda14ee4b2fa12b1d62a14c6d85)

### Expected result

Linux(Generic) completes job successfully

### What happens?

*Linux(Generic)* job [fails](https://github.com/kaomoneus/OpenTTD/actions/runs/6441089216/job/17490591433) with error:

```
  error: building dbus:x64-linux failed with: BUILD_FAILED
  Elapsed time to handle dbus:x64-linux: 13 s
  Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
  Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+dbus
  You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[dbus]+Build+error&body=Copy+issue+body+from+%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
  
  Error: Process completed with exit code 1.
```

### Why it happens?
When cache is enabled CI uses prebuilt packages. And things go smooth.
Same time, if cache is disabled or dropped then CI tries to rebuild few packages ("Install vcpkg and dependencies" step). In particular it tries to rebuild *sdl2* package. The latter triggers build of *dbus* package.

dbus 1.15 requires X11_xdmcp library. And even though it is installed by vcpkg, it is not found by subtle cmake command (see logs in attachment). Inability to build dbus with vcpkg might be considered as another non-OpenTTD (but vcpkg) issue.

### Should we fix dbus or OpenTTD?
I would answer "both".

* Perhaps it's worth [going to *vcpkg* project and fire an issue](https://github.com/microsoft/vcpkg/issues/new?title=[dbus]+Build+error&body=Copy+issue+body+from+%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md).
* From another side, this bug demonstrates the difference between modes
   * when cache is enabled and exists
   * ...and when cache doesn't exist or dropped or just disabled.

Caching is a good tool and improves build time dramatically. As its cons though we can't rely on cache too much, since it depends on external repos and thus makes our CI fragile. Keeping in mind that `release` pipeline is launched rarely perhaps it's worth disabling this tool for release?

Perhaps using Dockerfile with fixed versions of all dependencies would totally prevent such issues in future.

In scope of this bug pleas we could fix just OpenTTD CI (it must help in case if vcpkg cache will be lost).

Additionally consider firing vcpkg issue.

### Logs

[config-x64-linux-out.log](https://github.com/OpenTTD/OpenTTD/files/12837725/config-x64-linux-out.log)
[config-x64-linux-rel-CMakeCache.txt.log](https://github.com/OpenTTD/OpenTTD/files/12837726/config-x64-linux-rel-CMakeCache.txt.log)
[vcpkg-install-sdl2.log](https://github.com/OpenTTD/OpenTTD/files/12837727/vcpkg-install-sdl2.log)

### Steps to reproduce

1. Disable rust and vcpkg caches in release-linux.yml (assume modified workflow is in `master_no_cache` branch)
2. Launch `release` action for `master_no_cache` branch.
3. Watch for Linux(Generic) job

**Comment by glx22** [comment]
> Keeping in mind that `release` pipeline is launched rarely perhaps it's worth disabling this tool for release?

"rarely" like once a day to build nightly ?

**Comment by kaomoneus** [comment]
Created PR to demonstrate issue (see above)

**Comment by kaomoneus** [comment]
> "rarely" like once a day to build nightly ?
I think it is still "rarely" comparing to builds triggered by each PRs.
But I got your point. May be we could use for nightlies CIs with cache enabled, and for releases same but with cache disabled?

**Comment by TrueBrain** [comment]
What you mention really is an upstream issue with vcpkg, and not with us. You really should report the issue there.

One of the reasons we cache, is because we know from time to time vcpkg breaks; and although annoying, it is still worth using vcpkg, as that way we can drag in much more recent versions of our dependencies.

vcpkg doesn't allow pinning of versions, and they only have "latest". Sure, you could use an older commit in the vcpkg, but that is also not without risk. From time to time (and this happened more than once), upstream URLs change and vcpkg would fail in that case too. Basically, being on "latest" gives the least amount of risk.

We have upstreamed several issues and fixes to vcpkg over the last few months, and that will continue to happen.

Disabling cache for releases would be ill advised, as that means we only figure out such issues at the very last moment. Instead, once in a while there is a change that requires us to trash the cache, which most of the time results in us creating an upstream fix for vcpkg for it to get going again. This is just how the landscape currently is, and although vcpkg tests on many platforms, they all seem to be a bit newer than the ones we are targeting. So although annoying, currently I just assume vcpkg breaks from time to time.

The other issue with disabling cache: we would constantly be fixing vcpkg issues; and most issues resolve themselves over time anyway (as we are not the only ones using vcpkg). We do not have the resources available to constantly fix (or report) these kind of issues. Often it is better for us to just wait it out.

In short:

- Please report this problem with vcpkg; the problem cannot be fixed on our end.
- We know that vcpkg breaks from time to time; we either report those problems or fix them ourselves.

Sorry for the non-helpfulness, but I hope you understand the reasoning.

**Comment by kaomoneus** [comment]
@TrueBrain 

And yet I think you still need that missed dmcp dependency in your release-linux.yml. Consider case, when new sdl2 version will be published. Or just a case when something happend with your current cache (some github issues and so on). All those cases instantly broke your pipeline.

You're point is quite rational, but I would consider also having additional workflow just to check that you're cool on building everything from scratch (which is currently not).

Also you current state of mainstream is not forkable. Or let's say it's forkable but 'Release' workflow won't work on forked projects.

**Comment by kaomoneus** [comment]
Relevant *vcpkg* issue has been fired here: microsoft/vcpkg#34329
